client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,280 @@
+import { Controller } from '@hotwired/stimulus';
+
+const TAB_DEFAULTS = {
+  css: {
+    animate: 'animate-in',
+  },
+  keys: {
+    end: 'End',
+    home: 'Home',
+    left: 'ArrowLeft',
+    up: 'ArrowUp',
+    right: 'ArrowRight',
+    down: 'ArrowDown',
+  },
+  direction: {
+    ArrowLeft: -1,
+    ArrowRight: 1,
+  },
+}
+
+export class TabsController extends Controller<HTMLDivElement> {
+  static targets = ['tabList', 'tabTrigger', 'tabLabel', 'tabPanel'];
```

Comment:
Target names should not need the tab prefix. 

Otherwise the target will be

`data-w-tabs-target='tab-list`

It's much better to just be able to write

`data-w-tabs-target='list'`